### PR TITLE
ethclient/gethclient: fix bugs in override object encoding

### LINKED ISSUE #25615: gethclient call fails if the `Code` property is not present in the state overrides
#### System information

go-ethereum version: `v1.10.23`
OS & Version: OSX

#### Expected behaviour

While using the `gethclient/gethclient.go` client to override the state with the `gethclient/OverrideAccount` properties, if the `Code` property is not present it should be omitted or set to `null` on marshal.

```json
[
  {
    "data": "0xaa00000000000000000000000000000000000000",
    "from": "0xbb00000000000000000000000000000000000000",
    "to": "0xcc00000000000000000000000000000000000000"
  },
  "latest",
  {
    "0xdd00000000000000000000000000000000000000": {
      "nonce": "0x0",
      "code": null,
      "balance": null,
      "state": null,
      "stateDiff": null
    }
  }
]
```

#### Actual behaviour

If the `Code` property is not present in the overrides, it would still be marshalled into `0x` because of the type `hexutil.Bytes`.

```go
// Bytes marshals/unmarshals as a JSON string with 0x prefix.
// The empty slice marshals as "0x".
type Bytes []byte
```

The marshalled json:
```json
[
  {
    "data": "0xaa00000000000000000000000000000000000000",
    "from": "0xbb00000000000000000000000000000000000000",
    "to": "0xcc00000000000000000000000000000000000000"
  },
  "latest",
  {
    "0xdd00000000000000000000000000000000000000": {
      "nonce": "0x0",
      "code": "0x",
      "balance": null,
      "state": null,
      "stateDiff": null
    }
  }
]
```

Because of this behaviour, the request fails with the **execution reverted** error.

#### Steps to reproduce the behaviour

```go
// CallContract with override
override := gethclient.OverrideAccount{}

mapAcc := make(map[common.Address]gethclient.OverrideAccount)
mapAcc[address] = override

var err error
var result []byte

if result, err = client.CallContract(context.Background(), msg, nil, &mapAcc); err != nil {
	log.Fatalf("unexpected error: %v", err)
}
```

#### Backtrace

````
[backtrace]
````

When submitting logs: please submit them as text and not screenshots.

### PR DESCRIPTION
**Pull request by simonovic86**
If the `code` property is not present in the state overrides, it should be skipped or set to `null` on marshal. Because of the type of the property `hexutil.Bytes` the empty slice marshals to "0x" and the request fails.

fixes #25615

### CODE REVIEW

**Review by holiman — APPROVED** [review]
LGTM

File: ethclient/gethclient/gethclient_test.go [hunk 1]
```diff
@@ -327,3 +328,51 @@ func testCallContract(t *testing.T, client *rpc.Client) {
 		t.Fatalf("unexpected error: %v", err)
 	}
 }
+
+func TestOverrideAccountMarshal(t *testing.T) {
+	om := map[common.Address]OverrideAccount{
+		common.Address{0x11}: OverrideAccount{},
+		common.Address{0xaa}: OverrideAccount{
+			Nonce: 5,
+		},
```
[the remark this answers is not in the document]
Comment (fjl) [reply ethclient/gethclient/gethclient_test.go:337]: Well, we kind of already have that test because it's the same as not specifying the nonce at all...

[the remark this answers is not in the document]
Comment (holiman) [reply ethclient/gethclient/gethclient_test.go:337]: Yes right now it is -- and that's the behaviour that I want to be backed by test, so it doesn't change

[the remark this answers is not in the document]
Comment (fjl) [reply ethclient/gethclient/gethclient_test.go:337]: Hmm, maybe we misunderstood each other. I mean, since `0` is the zero value of `uint64`, it is already set to zero in every test except this one. If we add a test with `Nonce: 0` it will be exactly the same as the `{}` empty object test.

[the remark this answers is not in the document]
Comment (holiman) [reply ethclient/gethclient/gethclient_test.go:337]: >  If we add a test with Nonce: 0 it will be exactly the same as the OverrideAccount{} empty object test.

Yes, right now it is. But if we change `OverrideAccount` so that the types (including `Nonce`) are nillable, then the tests won't detect that change, just keep working. Whereas if you add an explicit `Nonce: 0`, then the test will flag the behavioural change.

[the remark this answers is not in the document]
Comment (holiman) [reply ethclient/gethclient/gethclient_test.go:341]: I pushed it
